### ISSUE TITLE
Too long command for rules execution when running on Windows

### ISSUE THREAD

**Issue by jvanoostveen** [SonarSource/SonarTS#193]
When I run the Sonar Scanner on our codebase, it crashes with the following error (partial):
```
Caused by: java.io.IOException: Cannot run program "node": CreateProcess error=206, The filename or extension is too long
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
        at org.sonar.plugin.typescript.ExternalTypescriptSensor.runRulesProcess(ExternalTypescriptSensor.java:218)
        ... 34 more
Caused by: java.io.IOException: CreateProcess error=206, The filename or extension is too long
        at java.lang.ProcessImpl.create(Native Method)
        at java.lang.ProcessImpl.<init>(ProcessImpl.java:386)
        at java.lang.ProcessImpl.start(ProcessImpl.java:137)
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
```

When I look at the command executed, I notice it contains every file it wants to pass to tslint, which are quite a few...
The length of the command (`node`) and the `tslint` call with the arguments and files is `33018` characters long! I recon this doesn't work on Windows...

I have setup the `sonar-project.properties` in such way, it contains several modules, each with their own `sonar.sources`.
When I limit the set of files for that specific module, it does succeed.
It might help to use this setting as a [glob for tslint](https://github.com/palantir/tslint/issues/692) when instantiating `tslint`.

Environment:
Windows 10 (build 1703)
PowerShell

**Comment by olivierlemasle** [comment]
Same issue (with Windows 2012R2)

**Comment by jvanoostveen** [comment]
Any news on this issue? It is really blocking any practical use of this plugin in SonarQube...

**Comment by jvanoostveen** [comment]
I'm looking forward to a fix :), I've been trying to find where it was called myself to make a quick fix/hack, but I haven't found it...

**Comment by rlouissidney** [comment]
+1 I have the same issue on a windows 10 Pro (Build 1715) Sonar 5.6.3

**Comment by vilchik-elena** [comment]
Fixed by #209

**Comment by inverno** [comment]
@jvanoostveen try the RC we just published, this is included
